### PR TITLE
Arxiv harvesters/normalizers

### PR DESCRIPTION
**Pull request by leb2dg**
## Purpose
Update Arxiv harvesters.

## Changes
* add harvester for non-OAI
* add normalizer for non-OAI
* update OAI harvester

### CODE REVIEW

**Comment by leb2dg** [comment]
@chrisseto, this is ready for review 🎉

File: providers/org/arxiv/apps.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+from .harvester import ArxivHarvester
 
 
-class AppConfig(OAIProviderAppConfig):
+class AppConfig(ProviderAppConfig):
     name = 'providers.org.arxiv'
     title = 'arxiv'
     long_title = 'ArXiv'
     home_page = 'https://arxiv.org'
-    rate_limit = (1, 3)
```
Comment (chrisseto) [inline providers/org/arxiv/apps.py]: Add the rate limit back in. It can be here or on the harvester

File: providers/org/arxiv/harvester.py [hunk 2]
```diff
@@ -1,7 +1,66 @@
...
+class ArxivHarvester(Harvester):
+    namespaces = {
+        'ns0': 'http://www.w3.org/2005/Atom'
+    }
+    start_page_num = 0
+    url = 'http://export.arxiv.org/api/query?search_query=all&max_results=100&sortBy=lastUpdatedDate&sortOrder=descending'
+
+    def get_next_url(self):
+        self.start_page_num += 100
+        return self.url + '&start={}'.format(self.start_page_num)
```
Comment (chrisseto) [inline providers/org/arxiv/harvester.py]: Use furl here just for consistency 
@caseyrollins has a pretty good example https://github.com/CenterForOpenScience/SHARE/pull/187/files#diff-7d27ce3f32de285c1155f8d2b7fc9734R22

File: providers/org/arxiv/harvester.py [hunk 3]
```diff
@@ -1,7 +1,66 @@
...
+        records = self.fetch_page(furl(url))
+
+        while True:
+            in_date_range = True
+            for index, record in enumerate(records):
+                # '2016-06-28T19:54:40Z'
+                updated = datetime.datetime.strptime(record.xpath('ns0:updated', namespaces=self.namespaces)[0].text, '%Y-%m-%dT%H:%M:%SZ').date()
+                if updated < start_date:
+                    logger.info('Record index {}: Updated record date {} is less than start date {}.'.format(index, updated, start_date))
+                    in_date_range = False
```
Comment (chrisseto) [inline providers/org/arxiv/harvester.py]: You can just `return` here, then you don't have to keep `in_date_range` updated

File: providers/org/arxiv/oai/apps.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from share.provider import OAIProviderAppConfig
+
+
+class AppConfig(OAIProviderAppConfig):
+    name = 'providers.org.arxiv.oai'
+    title = 'arxiv'
+    long_title = 'ArXiv'
+    home_page = 'https://arxiv.org'
+    rate_limit = (1, 3)
```
Comment (chrisseto) [inline providers/org/arxiv/oai/apps.py]: `rate_limit` only needs to be defined on/in one of appconfig or harvester.

File: providers/org/arxiv/harvester.py [hunk 5]
```diff
@@ -1,7 +1,66 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class ArxivHarvester(Harvester):
+    namespaces = {
+        'ns0': 'http://www.w3.org/2005/Atom'
+    }
+    start_page_num = 0
+    url = 'http://export.arxiv.org/api/query?search_query=all&max_results=100&sortBy=lastUpdatedDate&sortOrder=descending'
```
Comment (chrisseto) [inline providers/org/arxiv/harvester.py]: For niceness sake go ahead and just keep the url as `http://export.arxiv.org/api/query` and add the parameters using furl.

It looks like arxiv supports `https` but its not verified. Let's try to use SSL when available, even if its unverified its better than nothing.

File: providers/org/arxiv/harvester.py [hunk 6]
```diff
@@ -1,7 +1,66 @@
...
+from share import Harvester
+
+logger = logging.getLogger(__name__)
+
+
+class ArxivHarvester(Harvester):
+    namespaces = {
+        'ns0': 'http://www.w3.org/2005/Atom'
+    }
+    start_page_num = 0
```
Comment (chrisseto) [inline providers/org/arxiv/harvester.py]: As `start_page_num` is going to be instance specific define it inside of the `__init__`, but remember to call `super().__init__`

File: providers/org/arxiv/normalizer.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+from share.normalize import *  # noqa
+
+
+class Person(Parser):
```
Comment (chrisseto) [inline providers/org/arxiv/normalizer.py]: Arxiv also (sometimes) gives back affiliation information
```xml
    <author>
      <name>NAME</name>
      <arxiv:affiliation xmlns:arxiv="http://arxiv.org/schemas/atom">Department of Physics</arxiv:affiliation>
    </author>
    <author>
      <name>NAME</name>
      <arxiv:affiliation xmlns:arxiv="http://arxiv.org/schemas/atom">LETTERS, University College, PLACE</arxiv:affiliation>
    </author>
    <author>
      <name>NAME</name>
      <arxiv:affiliation xmlns:arxiv="http://arxiv.org/schemas/atom">Department of Physics</arxiv:affiliation>
    </author>
```

File: providers/org/arxiv/harvester.py (hunk 5 above)
Comment (leb2dg) [inline providers/org/arxiv/harvester.py]: Is it okay to set `verify=False`?
```
[2016-06-30 14:51:50,408][WARNING][py.warnings]: /Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py:821: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
```

Comment (chrisseto) [inline providers/org/arxiv/harvester.py]: For the moment yes, unverified https is better than just http.

File: providers/org/arxiv/normalizer.py (hunk 7 above)
Comment (leb2dg) [inline providers/org/arxiv/normalizer.py]: Is there a way to handle the affiliation not being there sometimes?

Comment (chrisseto) [inline providers/org/arxiv/normalizer.py]: Not currently, if you come up with a way to express it I'll show you how to extend the parser system.

**Comment by leb2dg** [comment]
@chrisseto, comments have been addressed. Sorry about the 50 merges... 👀

**Comment by leb2dg** [comment]
This is ready just waiting on what should be done with doi link.

**Comment by leb2dg** [comment]
@chrisseto, this is all ready and tested with the server 👍

**Pull request merged**